### ISSUE TITLE
Comments on `consult-notes-search-all`

### ISSUE THREAD

**Issue by protesilaos** [mclear-tools/consult-notes#7]
Hello again!

Brief comments on `consult-notes-search-all`:

- The `consult-notes-all-notes` declares `:type 'string`, though the doc
  string refers to a directory.  Should be it be `:type 'directory`
  instead?

- The default value of `consult-notes-all-notes` is an empty string.  On
  my end this seems to grep through all directories, instead of just
  those which contain notes.  I am not familiar with the `consult-grep`
  internals: perhaps it can accept a list of directories and collate the
  results?

- The default value of `consult-notes-grep-args` does not work on my end
  (GNU/Linux) because the `-S` flag is unavailable.  For reference, the
  `consult-grep-args` does not specify that flag.

**Comment by protesilaos** [comment]
> If there is a more elegant way to do this I'm happy to hear it!

Well, I am not the right person to answer this, but I'll try
nonetheless.  This works on my end:

```sh
~ $ grep -R "\<defcustom\>" --exclude-dir=".git" ~/Git/Projects/fontaine/ ~/Git/emacs-community/consult-notes/
```

Trying to pass a list of directories to `consult-grep` does not do the
trick.

**Comment by mclearc** [comment]
Hi Prot -- I've  revised the search in notes function (renamed to `consult-notes-search-in-all-notes`) so it should behave correctly. Note that I've also refactored the names of some of the variables (e.g. `consult-notes-data-dirs` is now `consult-notes-sources`), but hopefully modulo these changes it should work smoothly. I've also updated the docs to be a bit clearer about setting things up. Thanks again for your input!
